tss-esapi/tests/integration_tests/attributes_tests/session_attributes_tests.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use tss_esapi::{
+    attributes::{SessionAttributes, SessionAttributesBuilder, SessionAttributesMask},
+    tss2_esys::TPMA_SESSION,
+};
+
+macro_rules! test_valid_conversion {
+    ($tpm_value:expr, $method:ident) => {
+        let tpma_session_attribute: TPMA_SESSION = $tpm_value;
+        let session_attributes = SessionAttributes::from(tpma_session_attribute);
+        assert_eq!(
+            true,
+            session_attributes.$method(),
+            "SessionAttributes converted from TPMA_SESSION = {} did not produce the expected result with regard to {}.", tpma_session_attribute, stringify!($method),
+        );
+        assert_eq!(
+            tpma_session_attribute,
+            session_attributes.into(),
+            "Converting session attributes with {} set did not convert into the expected TPMA_SESSION value", std::stringify!($method),
+        );
+    };
+}
+
+macro_rules! test_valid_mask_conversion {
+    ($tpm_value:expr, $attribute:tt) => {
+        let tpma_session_attribute: TPMA_SESSION = $tpm_value;
+        let session_attributes_mask = SessionAttributesMask::from(tpma_session_attribute);
+        assert_eq!(
+            tpma_session_attribute,
+            session_attributes_mask.into(),
+            "Converting session attributes mask with {} set did not convert into the expected TPMA_SESSION value", $attribute,
+        );
+    };
+}
+
+#[test]
+fn test_valid_session_attributes_conversions() {
+    test_valid_conversion!(
+        1u8.checked_shl(0)
+            .expect("Failed to create 'continue session' value"),
+        continue_session
+    );
+    test_valid_conversion!(
+        1u8.checked_shl(1)
+            .expect("Failed to create 'audit exclusive' value"),
+        audit_exclusive
+    );
+    test_valid_conversion!(
+        1u8.checked_shl(2)
+            .expect("Failed to create 'audit reset' value"),
+        audit_reset
+    );
+    test_valid_conversion!(
+        1u8.checked_shl(5)
+            .expect("Failed to create 'decrypt' value"),
+        decrypt
+    );
+    test_valid_conversion!(
+        1u8.checked_shl(6)
+            .expect("Failed to create 'encrypt' value"),
+        encrypt
+    );
+    test_valid_conversion!(
+        1u8.checked_shl(7).expect("Failed to create 'audit' value"),
+        audit
+    );
+}
+
+#[test]
+fn test_invalid_session_attributes_conversions() {
```

Comment:
I wonder if this test can be implemented even if it's failing and then marked with [the `#[ignore]` attribute](https://doc.rust-lang.org/reference/attributes/testing.html#the-ignore-attribute).

Reply:
Yes, it should.

Reply:
Fixed